### DISCUSSION TITLE
Synchronously execution tasks in API routes

### DISCUSSION THREAD

**Discussion by enricoschaaf** [vercel/next.js#11661] [Help]
I want to send an email when a API route is hit, but I don't want to wait till the email is send to send a response. But when I don't await the response I get this error: Client network socket disconnected before secure TSL connection was established. I am using SendGrid/mail for this and I basically to this https://zeit.co/guides/deploying-nextjs-nodejs-and-sendgrid-with-zeit-now without await the response and it's a GraphQL API using Nexus.

**Comment by paulogdm** [comment]
Hello,

This is not possible. You need to wait for the "job" to finish before sending a response. Otherwise, the serverless function process will be frozen and no additional workload will be processed.

:heavy_check_mark: 
```
(req, res) => {
  // do work
  // send response
}
```

:negative_squared_cross_mark: 
```
(req, res) => {
  // do work
  // send response
  // do more work
}
```
